src/DataHydrator.php

```diff
@@ -186,15 +196,49 @@ private function resolveNode(array $data, array $params) {
                 throw new InvalidHydrateSpecException("The ${hydrateKey} must be a string. Instead got: $json");
             }
             $resolver = $this->getResolver($type);
-
-            $data = $resolver->resolve($data, $params);
+            $layoutCacheNodeKey = md5(json_encode($data));
+            $cacheData = $this->cache->getItem($layoutCacheNodeKey);
```

Comment:
We should probably add a `hasItem()` check in here. That could potentially save us a refetch in case the resolved value is null (which is a valid output).